packages/block-library/src/query-title/index.php

```diff
@@ -7,26 +7,32 @@
 
 /**
  * Renders the `core/query-title` block on the server.
- * For now it only supports Archive title,
+ * For now it supports Archive title, Search title and 404 title,
  * using queried object information
  *
  * @param array $attributes Block attributes.
  *
  * @return string Returns the query title based on the queried object.
  */
 function render_block_core_query_title( $attributes ) {
-	$type       = isset( $attributes['type'] ) ? $attributes['type'] : null;
+	$is_search  = is_search();
 	$is_archive = is_archive();
-	if ( ! $type || ( 'archive' === $type && ! $is_archive ) ) {
-		return '';
-	}
-	$title = '';
+	$title      = isset( $attributes['content'] ) ? $attributes['content'] : '';
 	if ( $is_archive ) {
 		$title = get_the_archive_title();
 	}
+	if ( $is_search ) {
```

Comment:
> Is it mostly because we want users to insert just one block (Query Title) in archive, search, and 404

The benefit I see is it means a simple theme can just have one index.html and the block will handle all scenarios...